File: VB/Q265259/Northwind.vb
Imports System
Imports DevExpress.Xpo
Namespace Northwind

    <Persistent("Products")> _
    Public Class Product
        Inherits XPLiteObject

        Private fProductID As Integer
        <Key(True)> _
        Public Property ProductID() As Integer
            Get
                Return fProductID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("ProductID", fProductID, value)
            End Set
        End Property
        Private fProductName As String
        <Size(40)> _
        Public Property ProductName() As String
            Get
                Return fProductName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("ProductName", fProductName, value)
            End Set
        End Property
        Private fQuantityPerUnit As String
        <Size(20)> _
        Public Property QuantityPerUnit() As String
            Get
                Return fQuantityPerUnit
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("QuantityPerUnit", fQuantityPerUnit, value)
            End Set
        End Property
        Private fUnitPrice As Decimal
        Public Property UnitPrice() As Decimal
            Get
                Return fUnitPrice
            End Get
            Set(ByVal value As Decimal)
                SetPropertyValue(Of Decimal)("UnitPrice", fUnitPrice, value)
            End Set
        End Property
        Private fUnitsInStock As Short
        Public Property UnitsInStock() As Short
            Get
                Return fUnitsInStock
            End Get
            Set(ByVal value As Short)
                SetPropertyValue(Of Short)("UnitsInStock", fUnitsInStock, value)
            End Set
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
    End Class

End Namespace

File: CS/Q265259/Northwind.cs
using System;
using DevExpress.Xpo;
namespace Northwind {

    [Persistent("Products")]
    public class Product :XPLiteObject {
        int fProductID;
        [Key(true)]
        public int ProductID {
            get { return fProductID; }
            set { SetPropertyValue<int>("ProductID", ref fProductID, value); }
        }
        string fProductName;
        [Size(40)]
        public string ProductName {
            get { return fProductName; }
            set { SetPropertyValue<string>("ProductName", ref fProductName, value); }
        }
        string fQuantityPerUnit;
        [Size(20)]
        public string QuantityPerUnit {
            get { return fQuantityPerUnit; }
            set { SetPropertyValue<string>("QuantityPerUnit", ref fQuantityPerUnit, value); }
        }
        decimal fUnitPrice;
        public decimal UnitPrice {
            get { return fUnitPrice; }
            set { SetPropertyValue<decimal>("UnitPrice", ref fUnitPrice, value); }
        }
        short fUnitsInStock;
        public short UnitsInStock {
            get { return fUnitsInStock; }
            set { SetPropertyValue<short>("UnitsInStock", ref fUnitsInStock, value); }
        }
        public Product(Session session) : base(session) { }
        public Product() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }



    [Persistent("Orders")]
    public class Order : XPLiteObject {
        int fOrderID;
        [Key(true)]
        public int OrderID {
            get { return fOrderID; }
            set { SetPropertyValue<int>("OrderID", ref fOrderID, value); }
        }
        string fShipName;
        [Size(40)]
        public string ShipName {
            get { return fShipName; }
            set { SetPropertyValue<string>("ShipName", ref fShipName, value); }
        }

        string fShipAddress;
         [Size(40)]
        public string ShipAddress {
            get { return fShipAddress; }
            set { SetPropertyValue<string>("ShipAddress", ref fShipAddress, value); }
        }
    
        public Order(Session session) : base(session) { }
        public Order() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }




}
